pyacq/devices/audio_pyaudio.py

```diff
@@ -93,12 +93,19 @@ def _configure(self, nb_channel=2, sample_rate=44100.,
                 raise ValueError(msg) from err
 
         self.output.spec['shape'] = (chunksize, self.nb_channel)
+        #~ self.output.spec['shape'] = (-1, self.nb_channel)
         self.output.spec['nb_channel'] = self.nb_channel
-        self.output.spec['dtype = '] = format
+        self.output.spec['dtype'] = format
         self.output.spec['sample_rate'] = float(int(self.sample_rate))
         gains = {'int16': 1./2**15, 'int32': 1./2**31, 'float32':1.}
         self.output.spec['gain'] = gains[self.format]
         self.output.spec['offset'] = 0.
+
+        self.input.spec['shape'] = (chunksize, self.nb_channel)
+        #~ self.input.spec['shape'] = (-1, self.nb_channel)
+        self.input.spec['nb_channel'] = self.nb_channel
```

Comment:
On this, I have change my opinion  many time. This reduduncy is unless but if you remove nb_channel from spec or params you have many time to do nb_channel = self.params['shape'][1].

At the first beginning nb_channel was not in stream_spec and one day I put it. I you prefer I can remove.
tell me I don't care.